src/services/permissions-checker.js

```diff
@@ -16,58 +16,144 @@ class PermissionsChecker {
 
   static expirationInSeconds = EXPIRATION_IN_SECONDS;
 
-  // This permissionsPerRendering is the cache, shared by all instances of PermissionsChecker.
-  static permissionsPerRendering = {};
+  static isrolesACLActivated;
+
+  // This permissions object is the cache, shared by all instances of PermissionsChecker.
+  static permissions = { collections: {}, renderings: {} };
+
 
   static cleanCache() {
-    PermissionsChecker.permissionsPerRendering = {};
+    PermissionsChecker.permissions = { collections: {}, renderings: {} };
   }
 
-  static getPermissions(renderingId) {
-    return PermissionsChecker.permissionsPerRendering[renderingId];
+  // In the teamACL format, all the permissions are stored by renderingId into "renderings".
+  // For the rolesACL format, the collections permissions are stored directly into "collections",
+  // and only their scopes are stored by renderingId into "renderings".
+  static getCollectionsPermissions(renderingId) {
+    if (PermissionsChecker.isrolesACLActivated) {
+      return PermissionsChecker.permissions.collections.data;
+    }
+    return PermissionsChecker.permissions.renderings[renderingId]
+      ? PermissionsChecker.permissions.renderings[renderingId].data
+      : null;
   }
 
-  static _setPermissionsInRendering(renderingId, permissions) {
-    PermissionsChecker.permissionsPerRendering[renderingId] = permissions;
+  _getScopePermissions(collectionName) {
+    if (PermissionsChecker.permissions.renderings[this.renderingId]
+      && PermissionsChecker.permissions.renderings[this.renderingId].data
+      && PermissionsChecker.permissions.renderings[this.renderingId].data[collectionName]) {
+      return PermissionsChecker.permissions.renderings[this.renderingId].data[collectionName].scope;
+    }
+    return null;
+  }
+
+  static transformPermissionsFromOldToNewFormat(permissions) {
+    Object.keys(permissions).forEach((modelName) => {
+      const { collection } = permissions[modelName];
+      permissions[modelName].collection = {
+        browseEnabled: collection.list || collection.searchToEdit,
+        readEnabled: collection.show,
+        editEnabled: collection.create,
+        addEnabled: collection.update,
+        deleteEnabled: collection.delete,
+        exportEnabled: collection.export,
+      };
+
+      if (permissions[modelName].actions) {
```

Comment:
You are using `permissions[modelName]` like a thousand times 😅  you should define a variable 😉